src/main/java/io/github/acm19/aws/interceptor/http/AwsRequestSigningApacheV5Interceptor.java

```diff
@@ -82,7 +82,9 @@ public void process(final HttpRequest request, final EntityDetails entityDetails
             }
         }
 
-        requestBuilder.headers(headerArrayToMap(request.getHeaders()));
+        Map<String, List<String>> headers = headerArrayToMap(request.getHeaders());
```

Comment:
Can we have a comment with a bit of context, like https://github.com/opensearch-project/opensearch-java/blob/main/java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java#L320. Possible a reference to the method `AbstractAws4Signer#doSign` in there.

Aside from that, I'd write it like, but it's personal preference, feel free to ignore:
```java
requestBuilder.headers(headerArrayToMap(request.getHeaders()));
requestBuilder.headers().put("x-amz-content-sha256", Collections.singletonList("required"));
```

Reply:
Looks cleaner, but `requestBuilder.headers()` is not a map you can modify

```
java.lang.UnsupportedOperationException
	at software.amazon.awssdk.utils.UnmodifiableMapOfLists.put(UnmodifiableMapOfLists.java:78)
	at software.amazon.awssdk.utils.UnmodifiableMapOfLists.put(UnmodifiableMapOfLists.java:35)
	at io.github.acm19.aws.interceptor.http.AwsRequestSigningApacheV5Interceptor.process(AwsRequestSigningApacheV5Interceptor.java:86)
	at io.github.acm19.aws.interceptor.http.AwsRequestSigningApacheV5InterceptorTest.testRepeatableEntity(AwsRequestSigningApacheV5InterceptorTest.java:102)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
```